src/Microsoft.ML.Transforms/PermutationFeatureImportance.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+
+namespace Microsoft.ML.Transforms
+{
+    using Stopwatch = System.Diagnostics.Stopwatch;
+
+    internal static class PermutationFeatureImportance<TResult>
+    {
+        public static ImmutableArray<TResult>
+            GetImportanceMetricsMatrix(
+                IHostEnvironment env,
+                ITransformer model,
+                IDataView data,
+                Func<IDataView, TResult> evaluationFunc,
+                Func<TResult, TResult, TResult> deltaFunc,
+                string features,
+                int topExamples)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(nameof(PermutationFeatureImportance<TResult>));
+            host.CheckValue(model, nameof(model));
+            host.CheckValue(data, nameof(data));
+            host.CheckValue(features, nameof(features));
```

Comment:
>CheckValue [](start = 17, length = 10)

CheckNonEmpty or CheckNonWhiteSpace. 
This is a string CheckValue would be happy to take "" but this would be invalid feature column name.